specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,575 @@
+          "enum": [
+            "Allow",
+            "Block",
+            "Log"
+          ],
+          "x-ms-enum": {
+            "name": "WebApplicationFirewallAction",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "Transform": {
+      "description": "Describes what transforms applied before matching",
+      "type": "string",
+      "enum": [
+        "Lowercase",
+        "Trim",
+        "UrlDecode",
+        "UrlEncode",
+        "RemoveNulls",
+        "HtmlEntityDecode"
+      ],
+      "x-ms-enum": {
+        "name": "WebApplicationFirewallTransform",
+        "modelAsString": true
+      }
+    },
+    "MatchVariable": {
+      "description": "Define match variables",
+      "required": [  
+        "name"
+      ],
+      "properties": {
+        "name": {
+          "description": "Match Variable",
+          "type": "string",
+          "enum": [
+            "RemoteAddr",
+            "RequestMethod",
+            "QueryString",
+            "PostArgs",
+            "RequestUri",
+            "RequestHeaders",
+            "RequestBody",
+            "RequestCookies"
+          ],
+          "x-ms-enum": {
+            "name": "WebApplicationFirewallMatchVariable",
+            "modelAsString": true
+          }
+        },
+        "selector": {
+          "description": "Describes field of the matchVariable collection",
+          "type": "string"
+        }
+      }
+    },
+    "MatchCondition": {
+      "description": "Define match conditions",
+      "required": [  
+        "matchVariables",  
+        "operator", 
+        "matchValues"
+      ],
+      "properties": {
+        "matchVariables": {
+          "description": "List of match variables",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MatchVariable"
+          }
+        },
+        "operator": {
+          "description": "Describes operator to be matched",
+          "type": "string",
+          "enum": [
+            "IPMatch",
+            "Equal",
+            "Contains",
+            "LessThan",
+            "GreaterThan",
+            "LessThanOrEqual",
+            "GreaterThanOrEqual",
+            "BeginsWith",
+            "EndsWith",
+            "Regex"
+          ],
+          "x-ms-enum": {
+            "name": "WebApplicationFirewallOperator",
+            "modelAsString": true
+          }
+        },
+        "negateCondition": {
+          "description": "Describes if this is negate condition or not",
+          "type": "boolean"
+        },
+        "matchValues": {
+          "description": "Match value",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "transforms": {
+          "description": "List of transforms",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Transform"
+          }
+        }
+      }
+    },
+    "ErrorResponse": {
```

Comment:
> @jianghaolu Its not exactly the same. So, skipping.

@meetug in that case please rename this definition to avoid collision with NetworkWatcher's `ErrorResponse` definition

Reply:
@number213 I am not exactly sure what you mean by NetworkWatcher but network.json does not have ErrorResponse.

Reply:
@meetug another top-level resource
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/networkWatcher.json#L1850